heart rate examples (#39)
deephaven/examples: fit/README.md

# Fit file as available from Strava

> Pull your own data from your Strava account. Visualize your data, manipulate your data, combine your data with other sources,  all while maintaining full ownership and control of the data in your own environment.

Reviewer: ```suggestion
Pull your own data from your Strava account. Visualize, manipulate, and combine your data with other sources, all while maintaining full ownership and control of the data in your own environment.
```

## Table of contents

>  * `ThursMorn.fit` - the data used to run this example

Reviewer: ```suggestion
 * `ThursMorn.fit` - The data used to run this example.
```

>  * `accessFit.py` - python script to demonstrate fit data inside deephaven 

Reviewer: ```suggestion
 * `accessFit.py` - Python script to demonstrate fit data inside Deephaven. 
```

>  * `*.png` - images used in this help

Reviewer: ```suggestion
 * `*.png` - Images used in this help.
```

Reviewer: In this help? or in these examples?

Author: I defer to whatever you / your team decides as the best final home for this chunk of work.

> ## Steps to run: example or your own data

Reviewer: ```suggestion
## Steps to run

These steps can be run with our example data or your own.
```

> 1. Login to your Strava account

Reviewer: ```suggestion
1. Login to your Strava account.
```

> 1. Select an activity.

Reviewer: these are all number 1

Author: Intentionally so... I thought that was the way to make markdown auto-number them.. so if you go add another step in the middle, all the steps after that still have the correct numbering.

If we don't like doing this stylistically, I'm happy to change it?

> 1. At the bottom of the menu, select the `...` for more

Reviewer: ```suggestion
1. At the bottom of the menu, select **...** for more.
```

   ![Strava export menu](StravaExport.png "Strava export menu")

> 1. Select `Export Original`. Provided you have supplied the data in fit format, this should now give a downloaded file with extension `.fit`

Reviewer: ```suggestion
1. Select **Export Original**. Provided you have supplied the data in fit format, this should now give a downloaded file with extension `.fit`
```

> 1. Put this `.fit` file in the data folder underneath your docker starting location:

Reviewer: ```suggestion
1. Put this `.fit` file in the data folder underneath your Docker starting location:
```

>    See: https://deephaven.io/core/docs/tutorials/launch-pre-built#set-up-your-deephaven-deployment

Reviewer: ```suggestion
   See our [Quickstart](https://deephaven.io/core/docs/tutorials/quickstart#set-up-your-deephaven-deployment) for further details.
```

Author: Wow, we renamed it, I was so confused.

   Add another directory level if desired to keep various projects segmented.

> 1. Ensure fitparse is setup (see below)

Reviewer: ```suggestion
1. Ensure fitparse is setup (see below).
```

> 1. Run the file given in a deephaven console: `accessFit.py`

Reviewer: ```suggestion
1. Run the given file in a Deephaven console: `accessFit.py`.
```

> 1. The table `heartRateData` should appear

Reviewer: ```suggestion
1. The `heartRateData` table should appear.
```

   ![Heart rate table](heartRateTable.png "Heart rate table")
1. Click the hamburger menu in the `heartRateData` tab.
1. Select `Chart Builder`
1. Select defaults (`Line`, `X-Axis=Timestamp`, `Series=Heart Rate`)

> 1. Press Create (You may need to scroll to see the button.)

Reviewer: ```suggestion
1. Press **Create** (note: you may need to scroll to see the button.)
```

> 1. A chart should appear

Reviewer: ```suggestion
1. A chart should appear:
```

   ![Heart rate chart](heartRateChart.png "Heart rate chart")

## Using fitparse module

> To use fitparse, install via your python session temporarily using:

Reviewer: ```suggestion
To use fitparse, install via your current Python session:
```

```python
import os
os.system("pip install fitparse")
```

> For more details on making the install persist between sessions, see:

Reviewer: ```suggestion
For more details on making the install persist between sessions, see [How to install Python packages](https://github.com/deephaven/deephaven.io/blob/main/core/docs/how-to-guides/install-python-packages.md).
```

> https://github.com/deephaven/deephaven.io/blob/main/core/docs/how-to-guides/install-python-packages.md

Reviewer: ```suggestion
```

Author: I still want the link to the full page. Is my linking style wrong.

If you would rather drop it, should I also drop the line above `For more details on making the install persist between sessions, see:`

> Note: YMMV if using a fit file which reports different data types. It appears that different sensors can report different data. In the example here, both GPS and heart rate monitor data is intertwined.

Reviewer: YMMV?

Author: https://www.howtogeek.com/693183/what-does-ymmv-mean-and-how-do-you-use-it/

Maybe there is a more appropriate alternative based on the "voice" we are using for this?

## Advanced challenge

> Some of the most interesting use cases for this data is to be able to correlate the data with other sources of interest to you.

Reviewer: ```suggestion
Some of the most interesting use cases for this data are to be able to correlate the data with other sources of interest to you.
```

> The first step is importing many fit files to compare:

Reviewer: ```suggestion
The first step is importing many fit files to compare:

```

> * todays heart rate with last months heart rate

Reviewer: ```suggestion
* Today's heart rate with last month's heart rate.
```

   ![Strava export menu](StravaExport.png "Strava export menu")
1. Select **Export Original**. Provided you have supplied the data in fit format, this should now give a downloaded file with extension `.fit`
1. Put this `.fit` file in the data folder underneath your Docker starting location:
   See our [Quickstart](https://deephaven.io/core/docs/tutorials/quickstart#set-up-your-deephaven-deployment) for further details.
   Add another directory level if desired to keep various projects segmented.
1. Ensure fitparse is setup (see below).
1. Run the given file in a Deephaven console: `accessFit.py`.
1. The `heartRateData` table should appear.
   ![Heart rate table](heartRateTable.png "Heart rate table")
1. Click the hamburger menu in the `heartRateData` tab.
1. Select `Chart Builder`
1. Select defaults (`Line`, `X-Axis=Timestamp`, `Series=Heart Rate`)
1. Press **Create** (note: you may need to scroll to see the button.)
1. A chart should appear:
   ![Heart rate chart](heartRateChart.png "Heart rate chart")

## Using fitparse module
To use fitparse, install via your current Python session:
```python
import os
os.system("pip install fitparse")
```

> For more details on making the install persist between sessions, see [How to install Python packages](https://github.com/deephaven/deephaven.io/blob/main/core/docs/how-to-guides/install-python-packages.md).

Reviewer 2: ```suggestion
For more details on making the install persist between sessions, see [How to install Python packages](https://github.com/deephaven/deephaven.io/blob/main/core/docs/how-to-guides/install-python-packages.md).
```
You only need the link once, remove second link below

Author: thanks good catch
